scripts/ci/lib.sh

```diff
@@ -68,6 +68,8 @@ ci_exit_trap() {
     done
 
     handle_dangling_processes
+
+    gate_flaky_tests "${exit_code}"
```

Comment:
Tbh. I'm not sure. It can be that this is used during test result import to mark job success or failure in DB. I would not change that. We still want to collect actual failures in DB.